test/parallel/test-tls-socket-close.js

```diff
@@ -41,18 +51,21 @@ function connectClient(server) {
       assert(serverTlsSocket);
 
       netSocket.destroy();
+      assert.strictEqual(netSocket.destroyed, true);
 
       setImmediate(() => {
-        assert.strictEqual(netSocket.destroyed, true);
-
+        assert.strictEqual(netSocketCloseEmitted, false);
```

Comment:
Why are we verifying netSocketCloseEmitted is false here and that serverTlsSocket.destroyed is false? Do we actually make those guarantees rather than just "we'll eventually destroy the tlsSocket?

Reply:
There is an "official" documentation on our website where we document that close callbacks are run after `setImmediate` callbacks.

Reply:
Would you mind adding a comment @lpinca?

Reply:
Done.

Reply:
> There is an "official" documentation on our website where we document that close callbacks are run after setImmediate callbacks.

Where?

Reply:
https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick

Reply:
Ungh I really grown to dislike the terminology we use in that document and "phases" since it almost never reflect what Node actually does. Thanks for finding the reference.